tests/test_cart.py

```diff
@@ -524,3 +535,171 @@ def test_cart_summary_page_empty_cart(client, request_cart):
     assert response.status_code == 200
     content = response.content.decode('utf-8')
     assert json.loads(content) == {}
+
+
+def test_product_group():
+    group = ProductGroup([Mock(is_shipping_required=Mock(return_value=False)),
+                          Mock(is_shipping_required=Mock(return_value=False))])
+    assert not group.is_shipping_required()
+    group = ProductGroup([Mock(is_shipping_required=Mock(return_value=True)),
+                          Mock(is_shipping_required=Mock(return_value=False))])
+    assert group.is_shipping_required()
+
+
+def test_cart_queryset(customer_user):
+    cart1 = Cart.objects.create(status=Cart.SAVED)
+    cart2 = Cart.objects.create(user=customer_user)
+
+    anonymous = Cart.objects.anonymous()
+
+    assert len(anonymous) == 1
+    assert anonymous[0] == cart1
+
+    saved = Cart.objects.saved()
+
+    assert len(saved) == 1
+    assert saved[0] == cart1
+
+    cart3 = Cart.objects.create(status=Cart.WAITING_FOR_PAYMENT)
+
+    waiting_for_payment = Cart.objects.waiting_for_payment()
+
+    assert len(waiting_for_payment) == 1
+    assert waiting_for_payment[0] == cart3
+
+    cart4 = Cart.objects.create(status=Cart.CHECKOUT)
+
+    checkout = Cart.objects.checkout()
+
+    assert len(checkout) == 1
+    assert checkout[0] == cart4
+
+    cart5 = Cart.objects.create(status=Cart.CANCELED)
+
+    canceled = Cart.objects.canceled()
+
+    assert len(canceled) == 1
+    assert canceled[0] == cart5
+
+    Cart.objects.save()
+
+    saved = Cart.objects.saved()
+
+    assert len(saved) == 5
+    assert len(Cart.objects.all()) == 5
+
+
+def test_get_user_open_cart(customer_user, opened_user_cart):
+
+    assert Cart.get_user_open_cart(customer_user) == opened_user_cart
+
+    cart = Cart.objects.create(user=customer_user)
+
+    assert Cart.get_user_open_cart(customer_user) == cart
+    assert Cart.objects.get(pk=opened_user_cart.pk).status == Cart.CANCELED
+
+
+def test_cart_repr():
+    cart = Cart()
+
+    assert cart.__repr__() == 'Cart(quantity=0)'
+
+    cart.quantity = 1
+
+    assert cart.__repr__() == 'Cart(quantity=1)'
+
+
+def test_cart_get_total_empty(db):
+    cart = Cart.objects.create()
+
+    with pytest.raises(AttributeError):
+        cart.get_total()
+
+
+def test_cart_change_user(customer_user):
+    cart1 = Cart.objects.create()
+
+    cart1.change_user(customer_user)
+    cart2 = Cart.objects.create()
+
+    cart2.change_user(customer_user)
+
+    old_cart = Cart.objects.get(pk=cart1.pk)
+
+    assert old_cart.user == customer_user
+    assert old_cart.status == Cart.CANCELED
+
+
+def test_cart_line_str__repr__(product_in_stock, request_cart_with_item):
```

Comment:
`repr` would be enough, use `repr()` instead of calling the `__repr__` magic method directly.